docs/0002-events-architecture.md

```diff
@@ -0,0 +1,84 @@
+# 2. Events Architecture
+
+Date: 2024-06-13
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, in the Visual Editor, Google Analytics (GA) tracking is implemented by hardcoding event data attributes directly into the component. This approach has limitations:
+
+- It tightly couples the GA tracking logic with the Visual Editor.
+- Any changes to GA tracking requirements necessitate changes in the Visual Editor component code.
+- The need to pass extra attributes for GA event data complicates the component logic.
+
+To address these issues, we propose implementing an Event Emitter system within the Visual Editor. This will allow us to decouple the GA tracking logic from the Visual Editor component. 
+Events will be emitted when users interact with the component (e.g., clicking a button or selecting an option from a dropdown). 
+These events can then be listened to by any part of the application to trigger GA data layer pushes or other actions, without requiring changes to the Visual Editor component itself.
```

Comment:
Can we make this clearer by saying "these events can be listened to by the Publishing Apps that imported the Visual Editor"